app/code/Magento/User/Api/Data/UserInterface.php

```diff
@@ -162,4 +162,19 @@ public function getInterfaceLocale();
      * @return $this
      */
     public function setInterfaceLocale($interfaceLocale);
+
+    /**
+     * Get user expiration date.
+     *
+     * @return string
+     */
+    public function getExpiresAt();
```

Comment:
There are couple of problems with this approach:
1. It's backward incompatible
2. Seems a bit weird to create this property in User module and only using it in Security module, these modules are interconnected enough

I recommend creating a new table with user_id as primary key and expires_at columns and do it within Security module then use it inside UserExpirationManager

Reply:
Should I put UserExpirationManager in the Security module as well?